assets/js/components/adminbar/AdminBarWidgets.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Admin Bar widgets.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import AdminBarUniqueVisitors from './AdminBarUniqueVisitors';
+import AdminBarSessions from './AdminBarSessions';
+import AdminBarImpressions from './AdminBarImpressions';
+import AdminBarClicks from './AdminBarClicks';
+import AnalyticsInactiveCTA from '../AnalyticsInactiveCTA';
+import CompleteModuleActivationCTA from '../CompleteModuleActivationCTA';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import { Row, Cell } from '../../material-components';
+import AdminBarZeroData from './AdminBarZeroData';
+const { useSelect } = Data;
+
+export default function AdminBarWidgets() {
+	const analyticsModuleConnected = useSelect( ( select ) => select( CORE_MODULES ).isModuleConnected( 'analytics' ) );
+	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
+
+	// True if _all_ admin bar sections have zero data.
+	const zeroData = useSelect( ( select ) => {
+		return AdminBarImpressions.selectHasZeroData( select ) &&
+			AdminBarClicks.selectHasZeroData( select ) &&
+			AdminBarUniqueVisitors.selectHasZeroData( select ) &&
+			AdminBarSessions.selectHasZeroData( select );
+	} );
```

Comment:
This approach works, but via #2252 we'll add a generally reusable approach to detect such a state to the plugin, which doesn't require splitting out the report definition in this specific way. To be clear, the currently used approach here is fine for where we're at right now in `develop`, but since #2252 will bring a common approach for that, we should go with that one. I think we should wait until that PR is merged and then apply that principle here:
* Rely on `getWidgetComponentProps()` to get scoped components (which includes `WidgetReportZero`) to pass to each of these four components.
    * While these four components are technically not widgets using the widget API, the only reason we didn't do that was because we didn't want to make the admin bar menu extensible _just yet_, as it might need further tweaks before. Therefore, we shouldn't register the components as widgets, but we can still use some of the infrastructure that the Widget API offers to us (more or less manually). Let's assign a specific slug to each of these four components (as if they were widgets), such as `adminBarImpressions`. This way we can use the `Widget*`-prefixed components like `WidgetReportZero`.
* Update the four components to return the received `WidgetReportZero` from props instead of the imported `ReportZero`. That component will set necessary state in the datastore (via `setWidgetState`).
* In `AdminBarWidgets` here, we need to call `getWidgetState` for each (active, see comment below) widget. If `state.Component` is `ReportZero` for every one, we know the admin bar is in `zeroData` state.
* We can then remove/revert the `selectHasZeroData` functions which will no longer be needed.

This approach is in line with how widgets that are _actually_ using the Widgets API work, and it will make an eventual transition to using the Widgets API in the admin bar smoother.

Reply:
I have a working version of this now, I'll make some of the further changes below then get it into CR.